packages/core/src/util/txInspector.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  Address,
+  CertificateType,
+  Ed25519KeyHash,
+  Lovelace,
+  PoolId,
+  StakeAddressCertificate,
+  StakeDelegationCertificate,
+  TokenMap,
+  TxAlonzo,
+  TxOut,
+  Value
+} from '../Cardano';
+
+const isMyAddress =
+  (addresses: Address[]) =>
+  ({ address }: { address: Address }) =>
+    addresses.includes(address!);
+
+const isOutgoing = (tx: TxAlonzo, ownAddresses: Address[]): boolean => tx.body.inputs.some(isMyAddress(ownAddresses));
```

Comment:
```suggestion
const isOutgoing = (tx: TxAlonzo, ownAddresses: Address[]): boolean => tx.body.inputs.some(isAddressWithin(ownAddresses));
```
... an _address within_ the wallet's _own addresses_  can be interpreted as "my address"